File: 0257-binary-tree-paths/0257-binary-tree-paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
var binaryTreePaths = function(root) {
    const res = []
    helper(root, "" + root.val)
    return res
    
    function helper(root, cur) {
        if (!root.left && !root.right) {
            return res.push(cur)
        }
        if (root.left) helper(root.left, cur + "->" + root.left.val)
        if (root.right) helper(root.right, cur + "->" + root.right.val)
    }
};

File: 0257-binary-tree-paths/0257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        def dfs(root: Optional[TreeNode]) -> List[str]:
            if not root:
                return []
            if not root.left and not root.right:
                return [str(root.val)]
            return [f'{root.val}->{sub}' for sub in dfs(root.left) + dfs(root.right)]
                
        return dfs(root)
            
                